File: algorithms_and_data_structures_1/alds1_1_b.py
def gcd(a, b):
    print("a: ", a, "b: ", b)
    if b > 0:
        gcd(b, a % b)
    else:
        print("ans", a)
        return a


if __name__ == "__main__":
    a, b = map(int, input().split())
    ans = gcd(a, b)
    print(ans)

File: algorithms_and_data_structures_1/alds1_1_b.cpp
#include <iostream>

int gcd(int a, int b)
{
    if (b == 0)
    {
        return a;
    }
    else
    {
        return gcd(b, (a % b));
    }
}

int main()
{
    //input
    int a, b;
    std::cin >> a >> b;

    int ans;
    ans = gcd(a, b);

    std::cout << ans << '\n';
}